tests/test_connection.py

```diff
@@ -288,6 +289,36 @@ def inner():
 
 
 @asyncio.coroutine
+def test_cancelled_connection_is_usable_asap(connect, loop):
+    fut = asyncio.Future(loop=loop)
+
+    @asyncio.coroutine
+    def inner():
+        fut.set_result(None)
+        yield from cur.execute("SELECT pg_sleep(10)")
+
+    conn = yield from connect()
+    cur = yield from conn.cursor()
+    task = ensure_future(inner(), loop=loop)
```

Comment:
Lets pass `cur` and `fut` explicitly as arguments, so it is easier to see where they come from